src/pspm_get_smr.m

```diff
@@ -0,0 +1,133 @@
+function [sts, import, sourceinfo] = pspm_get_smr(datafile, import)
+% ● Description
+%   pspm_get_smr is the main function for importting spike-smr files
+% ● Format
+%   [sts, import, sourceinfo] = pspm_get_smr(datafile, import);
+% ● Arguments
+%     import: [struct]
+%   .denoise: for marker channels in CED spike format (recorded as 'level'),
+%             filters out markers duration longer than the value given here (in
+%             ms).
+% ● History
+%   Introduced in PsPM 3.0
+%   Written in 2008-2015 by Dominik R Bach (Wellcome Trust Centre for Neuroimaging)
+%   Update in 2023 by Teddy
+
+%% 1 Initialise
+global settings
+if isempty(settings)
+  pspm_init;
+end
+sts = -1;
+sourceinfo = [];
+addpath(pspm_path('Import','SON'));
+%% 2 Get external file
+% using SON library
+warning off;
+% 2.1 Open file
+fid = fopen(datafile);
+% 2.2 Get channel list
+chanlist = SONChanList(fid);
+% 2.3 Preallocate memory for speed
+chandata = cell(numel(chanlist), 1);
+errorflag = [];
+% 2.4 Read channels
+for channel = 1:numel(chanlist)
+  try
+    [chandata{channel}, chanhead{channel}] = SONGetChannel(fid, chanlist(channel).number, 'milliseconds');
+  catch
+    errorflag(channel)=1;
+    chandata{channel}=[];
+    chanhead{channel}.title='';
+  end
+end
+fclose(fid);
+% 2.5 delete empty channels
+if ~isempty(errorflag)
+  ind=find(errorflag);
+  for channel=ind(end:-1:1)
+    chandata(channel)=[];
+    chanhead(channel)=[];
+  end
+end
+warning on;
+%% 3 extract individual channels
+% 3.1 loop through import jobs
+for iImport = 1:numel(import)
+  % 3.1.1 define channel number ---
+  if import{iImport}.channel > 0
+    channel = import{iImport}.channel;
+  else
+    channel = pspm_find_channel(arrayfun(@(i) chanhead{i}.title, 1:numel(chanhead), 'UniformOutput', 0), ...
+      import{iImport}.type); % bring channel names into a cell array
+    if channel < 1, return; end
+  end
+  if channel > numel(chandata)
+    warning('ID:channel_not_contained_in_file', ...
+      'Channel %02.0f not contained in file %s.\n', channel, datafile); 
+    return; 
+  end
+  sourceinfo.channel{iImport, 1} = sprintf('Channel %02.0f: %s', channel, chanhead{channel}.title);
+  % 3.1.2 convert to waveform or get sample rate for wave channel types
+  if strcmpi(settings.channeltypes(import{iImport}.typeno).data, 'wave')
+    if chanhead{channel}.kind == 1 % waveform
+      import{iImport}.data = chandata{channel};
+      import{iImport}.sr   = 1./chanhead{channel}.sampleinterval;
+    elseif chanhead{channel}.kind == 3 % timestamps
+      % get minimum frequency for reporting resolution
+      import{iImport}.minfreq = min(1./diff(chandata{channel}))*1000;
```

Comment:
For `minfreq`, there are two lines: line 78 and 82. I am unsure if they are repeating lines?